src/utils/SecurityUtils.js

```diff
@@ -0,0 +1,231 @@
+  if (resourceSecurity.accessControlList != null) {
+    const acl = resourceSecurity.accessControlList;
+    if (Array.isArray(acl)) {
+      const specificUserSecurity = acl.find((aclUser) => aclUser.id === userIdentifier);
+      if (specificUserSecurity !== undefined) {
+        return specificUserSecurity.role;
+      }
+    }
+  }
+  // If user is not specifically in ACL, return the default role of the resource
+  return resourceSecurity.default ?? null;
+};
+
+const getUserPermissionsForResource = (resourceSecurity, userIdentifier, resourceRolesToPermissionsMapping) => {
+  if (resourceSecurity == null) {
+    console.warn("Resource security is null or undefined, can't retrieve user permissions.");
+    return [];
+  }
+  if (resourceRolesToPermissionsMapping == null) {
+    console.warn("Mapping between roles and permissions is null or undefined, can't retrieve user permissions.");
+    return [];
+  }
+  const userRoleForResource = getUserRoleForResource(resourceSecurity, userIdentifier);
+  if (userRoleForResource === null) return [];
+  return getPermissionsFromRole(userRoleForResource, resourceRolesToPermissionsMapping);
+};
+
+const _getRolesFromMapping = (permissionsMapping) => {
+  return Object.keys(permissionsMapping);
+};
+
+const _getPermissionsFromMapping = (permissionsMapping) => {
+  const permissionsSet = new Set();
+  Object.values(permissionsMapping).forEach((permissions) =>
+    permissions.forEach((permission) => permissionsSet.add(permission))
+  );
+  return Array.from(permissionsSet);
+};
+
+// Example of format for orgzaniation permissions:
+// [
+//   {
+//     component: 'organization',
+//     roles: {
+//       none: [],
+//       viewer: ['read', 'read_security'],
+//       user: ['read', 'read_security', 'create_children'],
+//     },
+//   },
+//   {
+//     component: 'scenario',
+//     roles: { ... },
+//   },
+// ];
+const parseOrganizationPermissions = (organizationPermissions, addNoneRole = false) => {
+  if (organizationPermissions == null) {
+    console.warn("Organization permissions value is null or undefined, can't parse it.");
+    return null;
+  }
+  if (!Array.isArray(organizationPermissions)) {
+    console.error("Organization permissions value is not an array, can't parse it.");
+    return null;
+  }
+
+  const roles = {};
+  const permissions = {};
+  const permissionsMapping = {};
+
+  for (const componentPermissions of organizationPermissions) {
+    const componentKey = componentPermissions.component;
+    const mapping = componentPermissions.roles;
+    roles[componentKey] = _getRolesFromMapping(mapping);
+    permissions[componentKey] = _getPermissionsFromMapping(mapping);
+    permissionsMapping[componentKey] = mapping;
+  }
+
+  if (addNoneRole === true) {
+    for (const componentPermissions of organizationPermissions) {
+      const componentKey = componentPermissions.component;
+      if (roles[componentKey].indexOf('none') === -1) {
```

Comment:
you can use includes here instead of indexOf.
A proposal of writing by using && operator > !roles[componentKey].includes('none') && roles[componentKey].unshift('none');